### ISSUE TITLE
Proposal for new rule: forbid surrounding spaces

### ISSUE THREAD

**Issue by acalvo** [linthtml/linthtml#148]
Speaking with a colleague about other matters, we came up with a possible new linting rule that it'd be useful at least for us: forbidding spaces just after a tag is opened and just before a tag is closed. Line breaks should be handled, though. Here there are some valid/invalid code snippets to reflect the proposal and inspire a test suite for it.

**VALID** snippets:

```html
<h1>Lorem ipsum</h1>
```

```html
<p><strong>Lorem</strong> ipsum</p>
```

```html
<p><!-- comment --><strong>Lorem</strong> ipsum</p>
```

```html
<div>
    <p>Lorem ipsum dolor sit amet...</p>
</div>
```

```html
<div>
    <p>
        Lorem ipsum dolor sit amet...
    </p>
</div>
```

```html
<div>
    <p>
        <!-- comment -->
        Lorem ipsum dolor sit amet...
        <!-- comment -->
    </p>
</div>
```

```html
<div></div>
```

**INVALID** snippets:

```html
<h1> Lorem ipsum</h1>
```

```html
<h1>Lorem ipsum   </h1>
```

```html
<h1> Lorem ipsum </h1>
```

```html
<p> <strong>Lorem</strong> ipsum</p>
```

```html
<p> <!-- comment --><strong>Lorem</strong> ipsum</p>
```

```html
<div> </div>
```

**Comment by KamiKillertO** [comment]
That's a good idea.
:thinking: I'm wondering if we should split in two rules or not.
One rule to deal with space after an opening tab and one to deal with space before a closing tag

**Comment by acalvo** [comment]
Hmmm... Good point. After thinking about it, I wouldn't split it. Mainly because of two reasons. The first one is that if they were 2 separate rules, I don't see why someone would want to enable one and disable the other. The second one is that I looked for eslint/stylelint rules that could be similar to this case to see what they're doing and found this example: https://eslint.org/docs/rules/space-in-parens . It doesn't even expose a config to enable spaces in just one of the sides. You either allow them `"always"` or `"never"`.

But up to you! :)

**Comment by KamiKillertO** [comment]
:thinking: good point, I agree.
This is rule is not really suitable for a split. A more suitable rule for a split would be one who enforcing newlines.

**Comment by KamiKillertO** [comment]
@acalvo There's an opened PR (#226) to add this rule, please have a look

**Comment by acalvo** [comment]
Looks great @KamiKillertO !! 😀

Is there an easy way to test it against my codebase to see if it reports any oddity? To give you proper feedback before merging the PR, I mean.

**Comment by KamiKillertO** [comment]
Yes just add `"@linthtml/linthtml": "github:linthtml/linthtml#feat/rule_no_surrounding_whitespace",` in your `package.json`

**Comment by acalvo** [comment]
> Yes just add `"@linthtml/linthtml": "github:linthtml/linthtml#feat/rule_no_surrounding_whitespace",` in your `package.json`

Good trick, didn't know it. Thanks!

**Comment by KamiKillertO** [comment]
I've released [LintHTML 0.5.0](https://github.com/linthtml/linthtml/releases/tag/0.5.0) which contain this new rule.
